adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/IngOauth2Service.java

```diff
@@ -25,16 +25,28 @@ public IngOauth2Service(Oauth2Api oauth2Api,
     }
 
     public URI getAuthorizationRequestUri(Oauth2Service.Parameters parameters)  {
+        ParametersValidationService.validateScope(parameters);
         ClientAuthentication clientAuthentication =
             clientAuthenticationFactory.newClientAuthentication(getApplicationToken());
-        AuthorizationURLResponse authorizationUrlResponse = oauth2Api.getAuthorizationUrl(clientAuthentication)
+        AuthorizationURLResponse authorizationUrlResponse = oauth2Api.getAuthorizationUrl(clientAuthentication, parameters.getScope())
             .getBody();
 
         parameters.setClientId(getClientId());
+        clearParameters(parameters);
 
         return URI.create(StringUri.withQuery(authorizationUrlResponse.getLocation(), parameters.asMap()));
     }
 
+    private void clearParameters(Oauth2Service.Parameters parameters) {
+        if (parameters.get("limit") != null) {
+            parameters.remove("limit");
+        }
+
+        if (parameters.get("balanceTypes") != null) {
```

Comment:
i suggest to use `StringUtils.isBlank` for checking

Reply:
done